File: ARETE Global Interview/Abdallah Hemdan Interview/CompanySyetem.cpp
#include <iostream>
#include <vector>

using namespace std;

class Employee{
private:
  string id;
  int salary;
public:
  Employee(string id, int salary){
    this->id = id;
    this->salary = salary;
  }

  string getMyId(){
    return this->id;
  }

  int getMySalary(){
    return this->salary;
  }
}

class Hr{
private:
  string id;
  int salary;
public:
  Hr(string id, int salary){
    this->id = id;
    this->salary = salary;
  }

  string getMyId(){
    return this->id;
  }

  int getMySalary(){
    return this->salary;
  }
}

class Manager {
private:
    string managerId;
    int salay;
puplic:
  Manger(string managId, int salary){
    this->managerId = managId;
    this->salay = salary;
  }
  Employee createEmployee(string emploId, int emploSalary){
    Employee employee = new Employee(emploId,emploSalary);
    return employee;
  }
  Hr createHr(string hrId, int hrSalary) {
    Hr hr = new Hr(hrId,hrSalary);
    return hr;
  }
  Employee setEmployeeSalary(vector<Employee>& listOfEmployees, string id, int salary){
    listOfEmployee.push()
  }
}

int main(){
  vector<Employees> listOfEmployee;
  vector<Hr> listOfHrs;
  Manager ourManger = Manger(0, 50000);
  Employee firstEmpl = Manger.createEmployee(curCount, 2000);
  Hr firstGr = Manger.createHr(curCount, 3000);
  firstEmpl = Manager.setEmployeeSalary()
}

File: ARETE Global Interview/Abdallah Hemdan Interview/CompanySyetem.js
class Employee{
  // ...
  id
  salary
}
class Hr{
  // ...
  id
  salary
}


class Manger{
  getSalary = (employeeIdOrHrId, id) => {
   axios.get('endPoint')
    .then(){
      // ...
    }.catch{
      // ..
    }
  }
  
  setSalary = (salary,isEmployeeOrHr, id) =>{
    const dataToSent = {
      isEmployeeOrHr,
      id,
      salary
    } 
    axios.set('endPoint', dataToSent)
    .then(){
      // ...
    }.catch{
      // ..
    }
    
  }
  createEmployee = (id) => {
    const newEmployee = Employee(id, salary);
  }
  createHr = (id) => {
    const newHr = Hr(id, salary);
  }
}


const ourManger = Manger();
const newEmployee = ourManger.createEmployee(curCounter, salary);
